fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -172,7 +174,31 @@ def _call_oracle(self, lambda_vec):
             estimator = clone(estimator=self.estimator, safe=False)
 
         oracle_call_start_time = time()
-        estimator.fit(self.constraints.X, redY, **{self.sample_weight_name: redW})
+
+        # The line that is being replaced
+        # estimator.fit(self.constraints.X, redY, **{self.sample_weight_name: redW})
+
+        # ------- change made starting from here -------- #
+
+        # historical version: try and except, downside: too expensive
```

Comment:
In general, you don't need to put things like this into a comment.

For this specific case, I would ask for a quantification of 'too expensive' but... a general rule to follow is that exceptions should be truly exceptional anyway :-)